google/cloud/pubsublite/internal/partition_publisher.cc

```diff
@@ -170,6 +182,24 @@ void PartitionPublisher::Read() {
       });
 }
 
+void PartitionPublisher::OnReadStart() {
+  std::lock_guard<std::mutex> g{mu_};
+  if (!reading_) {
+    reading_.emplace();
+  }
+}
+
+void PartitionPublisher::OnReadEnd() {
+  absl::optional<promise<void>> read_done;
+  {
+    std::lock_guard<std::mutex> g{mu_};
+    read_done.swap(reading_);
+  }
+  if (read_done) {
```

Comment:
Same nit, we tend to write `if (read_done) read_done->set_value();`

Reply:
Done